### PR TITLE
Add big "unit test" in symbolic logic

### PR DESCRIPTION
**Pull request by rebcabin**
xs = {1, 2, 3}; Table[x*x, xs]

works in Mathematica, but not in mathics. I fixed it. In fact, the old code in 'lists.py' had a test of `head == 'Range'`, which would never have worked; it should have been `System<backtick>Range`. Because Ranges eval to Lists, anyway, my fix works for the `Range` case. I did not add unit tests for `Table`, but I also didn't find any in the existing test corpus.

I added a big file of symbolic logic: "GriesSchneider.m". It amounts to a serious unit test. The only difference to Mathematica I found (other than the bug above) was that mathics lacks `With`. This is easy to work around via `Module`. 

I also disabled some unit tests for console and launching because those unit tests cannot work inside PyCharm without some deeper changes.

### CODE REVIEW

**Comment by wolfv** [comment]
Hi @rebcabin do you have time to rebase your PR and squash some commits? 

That would be awesome.

File: README.rst [hunk 1]
```diff
@@ -11,6 +11,11 @@ Installing
 ----------
 `Here <https://github.com/mathics/Mathics/wiki/Installing>`_ is a short guide on how to install Mathics on your computer.
 
+Docker
```
Comment (GarkGarcia) [inline README.rst:14]: This is completely unrelated to the PR. If you'd like to address this please create a separate PR.

File: mathics/core/pattern.py [hunk 2]
```diff
@@ -289,27 +289,28 @@ def per_expr(yield_expr, expr_groups, sum=0):
 
                         if expr_groups:
                             expr, count = expr_groups.popitem()
-                            max_per_pattern = count / len(patterns)
+                            max_per_pattern = int(count / len(patterns))
```
Comment (GarkGarcia) [inline mathics/core/pattern.py:292]: Again, this isn't related to PR.

File: mathics/packages/GS.org [hunk 3]
```diff
@@ -0,0 +1,117 @@
+#+TITLE: CALCULATIONAL LOGIC
```
Comment (GarkGarcia) [inline mathics/packages/GS.org:1]: There's no need to include the .org file in the repository.

File: mathics/packages/mve.el [hunk 4]
```diff
@@ -0,0 +1,150 @@
+I am able to collect output data from a process, but I don't understand
```
Comment (GarkGarcia) [inline mathics/packages/mve.el:1]: What is this?

File: mathics/packages/ob-mathics.el [hunk 5]
```diff
@@ -0,0 +1,137 @@
+;;; ob-mathics.el --- org-babel functions for mathics evaluation
+
+;; Copyright (C) Brian Beckman
```
Comment (GarkGarcia) [inline mathics/packages/ob-mathics.el:3]: You cannot claim the rights for the contents of the files you submit in a PR. They must be licensed according to [our licensing scheme](https://github.com/mathics/Mathics/blob/master/COPYING.txt).

Comment (GarkGarcia) [reply mathics/packages/ob-mathics.el:3]: Eventhought you are using GLPv3, we cannot accept code with the comment `Copyright (C) Brian Beckman`.

File: mathics/packages/ob-python-sample.xl [hunk 6]
```diff
@@ -0,0 +1,375 @@
+;;; ob-python.el --- Babel Functions for Python      -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2009-2018 Free Software Foundation, Inc.
```
Comment (GarkGarcia) [inline mathics/packages/ob-python-sample.xl:3]: What is this??

File: test/test_console.py [hunk 7]
```diff
@@ -13,27 +13,31 @@
...
 
     def readline(self):
         return self.console.readline().decode('utf-8')
 
+    @unittest.skip
     def testLaunch(self):
         cons = self.console
 
-        self.assertRegexpMatches(self.readline(), '\r\n')
+        self.assertRegex(self.readline(), '\r\n')
```
Comment (GarkGarcia) [inline test/test_console.py:29]: This was already addressed by https://github.com/mathics/Mathics/pull/833. Again, this changes are not related to the PR.

**Comment by GarkGarcia** [comment]
@rebcabin I could not find any information in the about how `Table[x, {1, 2, 3}]` is supposed to work in the [Wolfram documentation](https://reference.wolfram.com/language/ref/Table.html?q=Table). Are you sure this works in Mathematica?

**Comment by mmatera** [comment]
The proper way to write that in mma would be

Table[x ,{x,{1,2,3,4}}]

**Comment by mmatera** [comment]
Table[x,{1,2,3}]  raise the error
"Table::itraw: Raw object HoldForm[1] cannot be used as an iterator."

**Comment by GarkGarcia** [comment]
> The proper way to write that in mma would be
> 
> Table[x ,{x,{1,2,3,4}}]

Ohh, I see. We should probably create another PR to address this though, since this one is filled with unrelated changes.

**Comment by GarkGarcia** [comment]
> Table[x,{1,2,3}] raise the error
> "Table::itraw: Raw object HoldForm[1] cannot be used as an iterator."

It looks like this has been fixed already:

```
Mathics 1.1.dev0
on CPython 3.6.9 (default, Jul 17 2020, 12:50:27)
using SymPy 1.6.2, mpmath 1.1.0

Copyright (C) 2011-2020 The Mathics Team.
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions.
See the documentation for the full license.

Quit by pressing CONTROL-D

In[1]:= Table[x ,{x,{1,2,3,4}}]
Out[1]= {1, 2, 3, 4}

In[2]:= Table[x ^ x ,{x,{1,2,3,4}}]
Out[2]= {1, 4, 27, 256}
```

**Comment by GarkGarcia** [comment]
I'll leave the PR opened for now since we could still leverage the unit tests for symbolic logic.

File: mathics/core/pattern.py (hunk 2 above)
Comment (rocky) [reply mathics/core/pattern.py:292]: > Again, this isn't related to PR.

This was addressed in https://github.com/mathics/Mathics/pull/937. Instead of `int(... / ...)`  `//` was used.

**Comment by rocky** [comment]
@rebcabin I definitely would like to include interesting real tests like this for Mathic after the 1.1.0 release. 

(I'd also like to the the pattern matching bug fixed too, but it appears as though this gets deep into the pattern-matching mechianism. Help fixing this would be greatly appreciated.)

We are very underpowered and getting the code to this state in of itself has been a big challenge so any help you could offer would be greatly appreciated.

A problem that I see with this particular test is that it doesn't hook into some sorted CI process. 

I am sure Mathematica too has a way of doing unit testing too, and it would be great if we were compatible with that, however that works. 

Would you consider rewriting your test code so that it uses `Quit[]` so can integrate with CI better?

**Comment by rebcabin** [comment]
i'll start working on this again. my old pr should be abandoned and i'll due to a regression on my part in the pattern matcher and i will start a new PR.

**Comment by rebcabin** [comment]
starting from scratch:

in my fork, i did "git reset --hard upstream/master," with upstream's being mathics/Mathics. "make check" yields one failed test:

```
4885 Tests for 906 built-in symbols, 4884 passed, 1 failed, 0 skipped.
Failed:
  - $IterationLimit in Reference of built-in symbols / Evaluation

FAILED
make: *** [Makefile:49: doctest] Error 1
```

I won't consider this serious (unless I see in this discussion that I should). I will start adding back in my old "symbolic-logic" tests, honoring Rocky's suggestion about Quit[] for the sake of CI and getting rid of the other objections noted above. I will try to make a new PR as small as possible.

**Comment by rocky** [comment]
Many thanks. I am optimistic we can add the tests into the upcoming 1.1.x release. 

I guess we'll discuss this in the weekly meeting later today.
